lib/vnet/app_resolver.go

```diff
@@ -221,25 +211,9 @@ func (r *TCPAppResolver) clusterClientForAppFQDN(ctx context.Context, profileNam
 			continue
 		}
 		for _, zone := range clusterConfig.DNSZones {
-			if !isSubdomain(fqdn, zone) {
-				// The queried app fqdn is not a subdomain of this zone, skip it.
-				continue
-			}
-
-			// Found a matching cluster.
-
-			if zone == clusterConfig.ProxyPublicAddr {
-				// We don't need to validate a custom DNS zone if this is the proxy public address, this is a
-				// normal app public_addr.
+			if isDescendantSubdomain(fqdn, zone) {
 				return clusterClient, nil
 			}
-			// The queried app fqdn is a subdomain of this custom zone. Check if the zone is valid.
-			if err := r.customDNSZoneChecker.validate(ctx, clusterConfig.ClusterName, zone); err != nil {
-				// Return an error here since the FQDN does match this custom zone, but the zone failed to
-				// validate.
-				return nil, trace.Wrap(err, "validating custom DNS zone %q matching queried FQDN %q", zone, fqdn)
-			}
-			return clusterClient, nil
```

Comment:
This used to be basically "Given the FQDN of an app, if you find a matching custom DNS zone configured in the cluster, validate its TXT records and return the cluster client, unless it's a zone of the proxy service, then just skip the validation and return the client".

Now we do not care about the TXT records, so we just return the client if the FQDN on the app matches a zone for this cluster.